overlord/snapstate/snapmgr.go

```diff
@@ -274,35 +274,30 @@ func (snapst *SnapState) SetTryMode(active bool) {
 	}
 }
 
-func autherForUserID(st *state.State, userID int) (store.Authenticator, error) {
-	var auther store.Authenticator
-	if userID > 0 {
-		user, err := auth.User(st, userID)
-		if err != nil {
-			return nil, err
-		}
-		auther = user.Authenticator()
+func userFromUserID(st *state.State, userID int) (*auth.UserState, error) {
+	if userID <= 0 {
```

Comment:
A negative user id would be a bug here. We can pass that on to auth.User and let it break.